File: backend-csharp/Models/Kitchen.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SousChefBackend.Models;

public class Kitchen
{
    public int Id { get; set; } // 厨房ID还是用 int 没问题
    
    // 🔥 适配 User 表的主键类型 (long)
    public long UserId { get; set; } 
    public User? User { get; set; }

    // 三大金刚
    public List<InventoryItem> InventoryItems { get; set; } = new();
    public List<MyCookware> MyCookwares { get; set; } = new();
    public List<MySeasoning> MySeasonings { get; set; } = new();
}

File: backend-java/src/main/java/com/souschef/entity/Kitchen.java
package com.souschef.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "kitchens")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Kitchen {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Integer id;
    
    @Column(name = "user_id", nullable = false, unique = true)
    private Long userId;
    
    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", insertable = false, updatable = false)
    private User user;
    
    @OneToMany(mappedBy = "kitchen", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<InventoryItem> inventoryItems = new ArrayList<>();
    
    @OneToMany(mappedBy = "kitchen", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<MyCookware> myCookwares = new ArrayList<>();
    
    @OneToMany(mappedBy = "kitchen", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<MySeasoning> mySeasonings = new ArrayList<>();
}


